modules/apps/announcements/announcements-web/src/main/java/com/liferay/announcements/web/internal/display/context/DefaultAnnouncementsDisplayContext.java

```diff
@@ -419,6 +433,46 @@ public boolean isTabs1Visible() {
 		return false;
 	}
 
+	private LinkedHashMap<Long, long[]>
+			_combinedSegmentsEntryRoleAndAnnouncementScopes()
+		throws PortalException {
+
+		LinkedHashMap<Long, long[]> announcementScopes =
+			AnnouncementsUtil.getAnnouncementScopes(
+				_announcementsRequestHelper.getUser());
+
+		Layout layout = _announcementsRequestHelper.getLayout();
```

Comment:
Why need to create a variable here? I think we could use `_announcementsRequestHelper.getScopeGroupId() `and `_announcementsRequestHelper.getCompanyId()` directly. What do you think?